views/js/pciCreator/ims/mathEntryInteraction/runtime/mathEntryInteraction.js

```diff
@@ -32,7 +33,8 @@ define([
     $,
     _,
     event,
-    MathQuill
+    MathQuill,
+    MathJax
```

Comment:
I'm not that familiar with this PCI, but hasn't it always been the purpose of MathQuill to create, edit and render the math elements? Therefore, instead of adding a new external dependency (MathJax), shouldn't the rendering take place here with a `MQ` function? (`this.createMathStatic` perhaps, I'm not too sure what each one does).

I see that this bug is in the context of the prompt. If this PCI has access to its own prompt element, I suggest trying to target that with MathQuill first.

Reply:
MathQuill supports only Latex, but we use MathML. We could create an editing widget in MathQuill format, but that is not easy for me.


When user **edits** in Authoring, out usual renderer is used: [Math.js](https://github.com/oat-sa/tao-item-runner-qti-fe/blob/develop/src/qtiCommonRenderer/renderers/Math.js#L50) with the other CkEditor features.
If I could somehow use the usual renderer in the PCI prompt, it would be much more correct, but I don't know how. PCI prompt is saved as an html string, we would need to parse it, extract math elements and wire up this renderer.

![Screenshot_25](https://user-images.githubusercontent.com/38751932/179183915-fbc3a944-1d3a-4032-a6ec-ec7c9221ce5b.png)

Reply:
Thanks, I understand a bit better. But doesn't this mean that we would have to import MathJax into _every_ PCI we support which has a prompt?

Reply:
Yes; I don't know how many interactions use it, but in the ticket there's also _TextReaderInteraction_ mentioned. Moreover, there's also see _Table_ element mentioned, not only _Math_, so proper solution should really be doing something about prompt html rendering on a more global level. (Or not allowing to insert those in PCI prompt at all)